packages/caliper-fabric/lib/connector-versions/v1/FabricGateway.js

```diff
@@ -0,0 +1,441 @@
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+'use strict';
+const semver = require('semver');
+const { DefaultEventHandlerStrategies, DefaultQueryHandlerStrategies, Gateway } = require('fabric-network');
+const { ConnectorBase, CaliperUtils, TxStatus, ConfigUtil } = require('@hyperledger/caliper-core');
+const FabricConnectorContext = require('../../FabricConnectorContext');
+
+const logger = CaliperUtils.getLogger('connectors/v1/FabricGateway');
+
+const EventStrategies = {
+    msp_all : DefaultEventHandlerStrategies.MSPID_SCOPE_ALLFORTX,
+    msp_any : DefaultEventHandlerStrategies.MSPID_SCOPE_ANYFORTX,
+    network_all : DefaultEventHandlerStrategies.NETWORK_SCOPE_ALLFORTX,
+    network_any : DefaultEventHandlerStrategies.NETWORK_SCOPE_ANYFORTX,
+};
+
+const QueryStrategies = {
+    msp_single : DefaultQueryHandlerStrategies.MSPID_SCOPE_SINGLE,
+    msp_round_robin : DefaultQueryHandlerStrategies.MSPID_SCOPE_ROUND_ROBIN,
+};
+
+//////////////////////
+// TYPE DEFINITIONS //
+//////////////////////
+
+/**
+ * @typedef {Object} FabricRequestSettings
+ *
+ * @property {string} channel Required. The name of the channel whose contract should be invoked.
+ * @property {string} contractId Required. The name/ID of the contract whose function
+ *           should be invoked.
+ * @property {string} contractFunction Required. The name of the function that should be
+ *           invoked in the contract.
+ * @property {string[]} [contractArguments] Optional. The list of {string} arguments that should
+ *           be passed to the contract.
+ * @property {Map<string, Buffer>} [transientMap] Optional. The transient map that should be
+ *           passed to the contract.
+ * @property {string} [invokerMspId] Optional. The MspId of the invoker. Required if there are more than
+ *           1 organisation defined in the network configuration file
+ * @property {string} invokerIdentity Required. The name of the client who should invoke the
+ * @property {boolean} [readOnly] Optional. Indicates whether the request is a submit or evaluation.
+ *           contract. If an admin is needed, use the organization name prefixed with a # symbol.
+ * @property {string[]} [targetPeers] Optional. An array of endorsing
+ *           peer names as the targets of the invoke. When this
+ *           parameter is omitted the target list will include the endorsing peers assigned
+ *           to the target contract, or if it is also omitted, to the channel.
+ * @property {string[]} [targetOrganizations] Optional. An array of endorsing
+ *           organizations as the targets of the invoke. If both targetPeers and targetOrganizations
+ *           are specified then targetPeers will take precedence
+ */
+
+/////////////////////////////
+// END OF TYPE DEFINITIONS //
+/////////////////////////////
+
+/**
+ */
+class V1FabricGateway extends ConnectorBase {
```

Comment:
Do we want this to be named the same as the exported class?

Reply:
I was debating this with myself but would then see the need to split the connectors into 2 directories redundant (we would then also change the wallet names as well). I couldn't decide either so let it as is to see what comments came back :-)